--- a/lib/index.js
+++ b/lib/index.js
@@ -37,7 +37,8 @@
   var app = express();
   var server = _server || http.createServer();
   var proxyEvents = new EventEmitter();
-  var middlewares = ['./init', '../biz']
+  var middlewares = process.env.FLUXY_WHISTLE_RUNTIME ? ['./init'] : ['./init', '../biz'];
+  middlewares = middlewares
     .concat(require('./inspectors'))
     .concat(config.middlewares)
     .concat(require('./handlers'));
@@ -205,7 +206,8 @@
       socksServer.useAuth(socks.auth.None());
     });
   }
-  require('../biz/init')(proxyEvents, function () {
+  var init = process.env.FLUXY_WHISTLE_RUNTIME ? function (_, done) { done(); } : require('../biz/init');
+  init(proxyEvents, function () {
     server.on('request', app);
     execCallback();
   });
--- a/lib/config.js
+++ b/lib/config.js
@@ -410,6 +410,7 @@
 }
 
 function resolvePath(file) {
+  if (typeof file === 'function') return file;
   if (!file || !(file = file.trim())) {
     return file;
   }
@@ -958,6 +959,7 @@
     config.certDir = path.resolve(newConf.certDir);
   }
   var mw = newConf.middlewares || newConf.middleware;
+  if (Array.isArray(mw)) config.middlewares = mw;
   if (typeof mw == 'string') {
     config.middlewares = mw.trim().split(/\s*,\s*/);
   }
